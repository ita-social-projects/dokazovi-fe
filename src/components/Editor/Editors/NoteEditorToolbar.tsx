// import React from 'react';
// import { useTranslation } from 'react-i18next';
// import { langTokens } from '../../../locales/localizationInit';
// import ImageHandlersContainer from '../CustomModules/ImageHandlersContainer';
// import '../generalEditor.styles.css';
// import { IEditorToolbarProps } from '../types';

// const NoteEditorToolbar: React.FC<IEditorToolbarProps> = ({ editor }) => {
//   const { t } = useTranslation();

//   return (
//     <div id="toolbar">
//       <span className="ql-formats">
//         <select className="ql-header" defaultValue="">
//           <option value="3">{t(langTokens.editor.title)}</option>
//           <option value="4">{t(langTokens.editor.introduction)}</option>
//           <option value="5">{t(langTokens.editor.note)}</option>
//           <option value="">{t(langTokens.editor.plain)}</option>
//         </select>
//       </span>
//       <span className="ql-formats">
//         <button title="Bold" type="button" className="ql-bold" />
//         <button title="Italic" type="button" className="ql-italic" />
//         <button title="Underline" type="button" className="ql-underline" />
//       </span>
//       <span className="ql-formats">
//         <button
//           title="Ordered List"
//           type="button"
//           className="ql-list"
//           value="ordered"
//         />
//         <button
//           title="Unordered List"
//           type="button"
//           className="ql-list"
//           value="bullet"
//         />
//       </span>
//       <span className="ql-formats">
//         <button
//           title="Super"
//           type="button"
//           className="ql-script"
//           value="super"
//         />
//         <button title="Sub" type="button" className="ql-script" value="sub" />
//         <button title="Quote" type="button" className="ql-blockquote" />
//         <button type="button" className="ql-link" />
//       </span>
//       <ImageHandlersContainer editor={editor} />
//     </div>
//   );
// };

// export default NoteEditorToolbar;
export {};
