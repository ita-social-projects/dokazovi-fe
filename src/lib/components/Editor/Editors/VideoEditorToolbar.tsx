import React from 'react';
import '../generalEditor.styles.css';
import { IEditorToolbarProps } from '../types';

const VideoEditorToolbar: React.FC<IEditorToolbarProps> = () => {
  return (
    <div id="toolbar">
      <span className="ql-formats">
        <button title="Underline" type="button" className="ql-underline" />
        <button title="Bold" type="button" className="ql-bold" />
        <button title="Italic" type="button" className="ql-italic" />
      </span>
    </div>
  );
};

export default VideoEditorToolbar;
