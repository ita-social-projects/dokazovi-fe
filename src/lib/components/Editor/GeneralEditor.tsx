import React, { useEffect, useRef, useState } from 'react';
import Quill from 'quill';
import ReactQuill from 'react-quill';
import { modules, formats } from './utilities';
import 'react-quill/dist/quill.snow.css';
import { IEditorToolbarProps } from './types';

export interface IQuillEditorProps {
  toolbar: React.ComponentType<IEditorToolbarProps>;
  initialHtmlContent?: string;
  onHtmlContentChange: (htmlContent: string) => void;
  onTextContentChange?: (textContent: string) => void;
}

const Editor: React.FC<IQuillEditorProps> = (props) => {
  const {
    initialHtmlContent,
    onHtmlContentChange,
    onTextContentChange,
  } = props;
  const editorRef = useRef<ReactQuill>(null);
  const [editor, setEditor] = useState<Quill>();

  useEffect(() => {
    if (editorRef.current) {
      const editorResult = editorRef.current.getEditor();
      setEditor(editorResult);

      if (onTextContentChange) {
        editorResult.on('text-change', () => {
          onTextContentChange(editorResult.getText());
        });
      }
    }
  }, []);

  return (
    <div className="text-editor">
      <props.toolbar editor={editor} />
      <ReactQuill
        theme="snow"
        defaultValue={initialHtmlContent}
        onChange={onHtmlContentChange}
        placeholder="Write something awesome..."
        modules={modules}
        formats={formats}
        ref={editorRef}
      />
    </div>
  );
};

export const GeneralEditor = React.memo(Editor);
