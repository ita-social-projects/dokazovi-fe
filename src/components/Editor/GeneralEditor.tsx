import React, { useEffect, useRef, useState } from 'react';
import Quill from 'quill';
import ReactQuill from 'react-quill';
import { modules, formats } from './utilities';
import 'react-quill/dist/quill.snow.css';
import { EditorToolbar } from './Editors/EditorToolbar';

export interface IQuillEditorProps {
  isVideoPost?: boolean;
  initialHtmlContent?: string;
  onHtmlContentChange: (htmlContent: string) => void;
  onTextContentChange?: (textContent: string) => void;
}

const Editor: React.FC<IQuillEditorProps> = (props) => {
  const {
    initialHtmlContent,
    onHtmlContentChange,
    onTextContentChange,
    isVideoPost,
  } = props;
  const editorRef = useRef<ReactQuill>(null);
  const [editor, setEditor] = useState<Quill>();

  useEffect(() => {
    if (editorRef.current) {
      const editorResult = editorRef.current.getEditor();
      setEditor(editorResult);

      if (onTextContentChange) {
        editorResult.on('text-change', () => {
          onTextContentChange(editorResult.getText());
        });
      }
    }
  }, []);

  return (
    <div className="text-editor" data-testid="text-editor_test">
      <EditorToolbar editor={editor} isVideoPost={isVideoPost} />
      <ReactQuill
        theme="snow"
        defaultValue={initialHtmlContent}
        onChange={onHtmlContentChange}
        placeholder="Write something awesome..."
        modules={modules}
        formats={formats}
        ref={editorRef}
        scrollingContainer="html"
      />
    </div>
  );
};

export const GeneralEditor = React.memo(Editor);
