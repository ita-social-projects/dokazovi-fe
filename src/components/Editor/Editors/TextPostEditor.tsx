import React, { useState } from 'react';
import { Grid } from '@material-ui/core';
import { BorderBottom } from '../../../old/lib/components/Border';
import PreviewInput from '../PreviewInput';
import { GeneralEditor } from '../GeneralEditor';
import { PostPreviewCard } from '../../Posts/Cards/PostPreviewCard';
import { IPost } from '../../../old/lib/types';

interface ITextPostEditorProps {
  isVideoPost?: boolean;
  initialHtmlContent?: string;
  initialPreview: string;
  onHtmlContentChange: (value: string) => void;
  initialWasPreviewManuallyChanged: boolean;
  onPreviewManuallyChanged?: () => void;
  onPreviewChange: (value: string) => void;
  previewPost: IPost;
  disableAutoChanges?: () => void;
}

const PostEditor: React.FC<ITextPostEditorProps> = ({
  isVideoPost,
  initialHtmlContent,
  initialPreview,
  onHtmlContentChange,
  initialWasPreviewManuallyChanged,
  onPreviewManuallyChanged,
  onPreviewChange,
  previewPost,
  disableAutoChanges,
}) => {
  const [textContent, setTextContent] = useState<string>('');

  return (
    <>
      <GeneralEditor
        initialHtmlContent={initialHtmlContent}
        onHtmlContentChange={onHtmlContentChange}
        isVideoPost={isVideoPost}
        onTextContentChange={setTextContent}
      />
      <BorderBottom />
      <Grid container direction="row" alignItems="stretch">
        <Grid
          item
          container
          xs={12}
          lg={8}
          md={6}
          direction="column"
          alignItems="stretch"
        >
          <PreviewInput
            disableAutoChanges={disableAutoChanges}
            initialPreview={initialPreview}
            editorTextContent={
              !initialWasPreviewManuallyChanged ? textContent : undefined // optimizing rerenders when we don't need editor content
            }
            initialWasManuallyChanged={initialWasPreviewManuallyChanged}
            onManuallyChanged={onPreviewManuallyChanged}
            onPreviewChange={onPreviewChange}
          />
        </Grid>
        <Grid item xs={12} lg={4} md={6}>
          <PostPreviewCard post={previewPost} shouldNotUseLink />
        </Grid>
      </Grid>
    </>
  );
};

export const TextPostEditor = React.memo(PostEditor);
