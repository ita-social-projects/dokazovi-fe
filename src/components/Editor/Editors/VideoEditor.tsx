import React from 'react';
import { GeneralEditor } from '../GeneralEditor';

interface IVideoEditorProps {
  initialHtmlContent?: string;
  onHtmlContentChange: (content: string) => void;
}

const VideoEditor: React.FC<IVideoEditorProps> = ({
  initialHtmlContent,
  onHtmlContentChange,
}) => {
  return (
    <>
      <GeneralEditor
        initialHtmlContent={initialHtmlContent}
        onHtmlContentChange={onHtmlContentChange}
      />
    </>
  );
};

export default VideoEditor;
