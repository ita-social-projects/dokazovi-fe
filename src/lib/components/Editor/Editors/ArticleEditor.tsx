import React, { useEffect, useRef, useState } from 'react';
import ReactQuill from 'react-quill';
import Quill from 'quill';
import { PostTypeEnum } from '../../../types';
import { BorderBottom } from '../../Border';
import ContentPreviewContainer from '../ContentPreviewContainer';
import GeneralEditor from '../GeneralEditor';
import ArticleEditorToolbar from './ArticleEditorToolbar';

interface IArticleEditorProps {
  dispatchContent: (content: string) => void;
}

const ArticleEditor: React.FC<IArticleEditorProps> = ({ dispatchContent }) => {
  const [editor, setEditor] = useState<Quill>();
  const articleEditor = useRef<ReactQuill | null>(null);
  const [editorContent, setEditorContent] = useState<string>('');

  useEffect(() => {
    if (articleEditor.current) {
      setEditor(articleEditor.current.getEditor());
      setEditorContent(
        articleEditor.current.getEditor().getText().slice(0, -1), // slice removes additional \n added by Quill
      );
    }
  }, []);

  editor?.on('text-change', () => {
    setEditorContent(editor.getText());
  });

  return (
    <>
      <GeneralEditor
        type={PostTypeEnum.ARTICLE}
        dispatchContent={dispatchContent}
        toolbar={<ArticleEditorToolbar editor={editor} />}
        ref={articleEditor}
      />
      <BorderBottom />
      <ContentPreviewContainer
        previewText={editorContent}
        previewType={PostTypeEnum.ARTICLE}
        previewCardType="Стаття"
      />
    </>
  );
};

export default ArticleEditor;
