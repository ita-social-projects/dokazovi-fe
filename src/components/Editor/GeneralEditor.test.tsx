import React from 'react';
import { render, screen } from '@testing-library/react';
import { GeneralEditor } from './GeneralEditor';
import { VideoEditorToolbar } from './Editors/VideoEditorToolbar';
import { JSDOM } from 'jsdom';

const dom = new JSDOM();
global.document = dom.window.document;
global.document.execCommand = jest.fn((value) => value);
global.window = dom.window;

const mockHtmlContentCange = jest.fn((value) => value);
const mockTextContantChange = jest.fn((value) => value);
jest.mock(
  'quill-image-uploader',
  () =>
    function (quill, options) {
      var self = this;
      self.quill = quill;
      self.options = options;
      self.range = null;
      self.selectLocalImage = jest.fn((value) => value);
      self.handleDrop = jest.fn((value) => value);
      self.handlePaste = jest.fn((value) => value);
      self.readAndUploadFile = jest.fn((value) => value);
      self.fileChanged = jest.fn((value) => value);
      self.insertBase64Image = jest.fn((value) => value);
      self.insertToEditor = jest.fn((value) => value);
      self.removeBase64Image = jest.fn((value) => value);
    },
);

test('component renders with all props setted', () => {
  render(
    <GeneralEditor
      onHtmlContentChange={mockHtmlContentCange}
      onTextContentChange={mockTextContantChange}
      toolbar={VideoEditorToolbar}
      initialHtmlContent={`<p>test</p>`}
    />,
  );
  expect(screen.getByTestId('text-editor_test')).toBeInTheDocument();
  expect(screen.getByTestId('text-editor_test')).toMatchSnapshot();
});

test('component renders innitial HTML content', () => {
  render(
    <GeneralEditor
      onHtmlContentChange={mockHtmlContentCange}
      onTextContentChange={mockTextContantChange}
      toolbar={VideoEditorToolbar}
      initialHtmlContent={`<p>test111</p>`}
    />,
  );
  expect(screen.getByText('test111')).toBeInTheDocument();
});

test('component renders without non required props', () => {
  render(
    <GeneralEditor
      onHtmlContentChange={mockHtmlContentCange}
      toolbar={VideoEditorToolbar}
    />,
  );
  expect(screen.getByTestId('text-editor_test')).toBeInTheDocument();
  expect(screen.getByTestId('text-editor_test')).toMatchSnapshot();
});
