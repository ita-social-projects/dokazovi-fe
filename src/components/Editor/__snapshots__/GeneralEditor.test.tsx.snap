// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component renders with all props setted 1`] = `
<div
  class="text-editor"
  data-testid="text-editor_test"
>
  <div
    class="ql-toolbar ql-snow"
    id="toolbar"
  >
    <span
      class="ql-formats"
    >
      <span
        class="ql-header ql-picker"
      >
        <span
          aria-controls="ql-picker-options-0"
          aria-expanded="false"
          class="ql-picker-label"
          data-label="Звичайний"
          data-value=""
          role="button"
          tabindex="0"
        >
          <svg
            viewBox="0 0 18 18"
          >
             
            <polygon
              class="ql-stroke"
              points="7 11 9 13 11 11 7 11"
            />
             
            <polygon
              class="ql-stroke"
              points="7 7 9 5 11 7 7 7"
            />
             
          </svg>
        </span>
        <span
          aria-hidden="true"
          class="ql-picker-options"
          id="ql-picker-options-0"
          tabindex="-1"
        >
          <span
            class="ql-picker-item"
            data-label="Заголовок"
            data-value="3"
            role="button"
            tabindex="0"
          />
          <span
            class="ql-picker-item"
            data-label="Вступ"
            data-value="4"
            role="button"
            tabindex="0"
          />
          <span
            class="ql-picker-item"
            data-label="Примітка"
            data-value="5"
            role="button"
            tabindex="0"
          />
          <span
            class="ql-picker-item"
            data-label="Звичайний"
            data-value=""
            role="button"
            tabindex="0"
          />
        </span>
      </span>
      <select
        class="ql-header"
        style="display: none;"
      >
        <option
          value="3"
        >
          Заголовок
        </option>
        <option
          value="4"
        >
          Вступ
        </option>
        <option
          value="5"
        >
          Примітка
        </option>
        <option
          selected=""
          value=""
        >
          Звичайний
        </option>
      </select>
    </span>
    <span
      class="ql-formats"
    >
      <button
        class="ql-bold"
        title="Bold"
        type="button"
      >
        <svg
          viewBox="0 0 18 18"
        >
           
          <path
            class="ql-stroke"
            d="M5,4H9.5A2.5,2.5,0,0,1,12,6.5v0A2.5,2.5,0,0,1,9.5,9H5A0,0,0,0,1,5,9V4A0,0,0,0,1,5,4Z"
          />
           
          <path
            class="ql-stroke"
            d="M5,9h5.5A2.5,2.5,0,0,1,13,11.5v0A2.5,2.5,0,0,1,10.5,14H5a0,0,0,0,1,0,0V9A0,0,0,0,1,5,9Z"
          />
           
        </svg>
      </button>
      <button
        class="ql-italic"
        title="Italic"
        type="button"
      >
        <svg
          viewBox="0 0 18 18"
        >
           
          <line
            class="ql-stroke"
            x1="7"
            x2="13"
            y1="4"
            y2="4"
          />
           
          <line
            class="ql-stroke"
            x1="5"
            x2="11"
            y1="14"
            y2="14"
          />
           
          <line
            class="ql-stroke"
            x1="8"
            x2="10"
            y1="14"
            y2="4"
          />
           
        </svg>
      </button>
      <button
        class="ql-underline"
        title="Underline"
        type="button"
      >
        <svg
          viewBox="0 0 18 18"
        >
           
          <path
            class="ql-stroke"
            d="M5,3V9a4.012,4.012,0,0,0,4,4H9a4.012,4.012,0,0,0,4-4V3"
          />
           
          <rect
            class="ql-fill"
            height="1"
            rx="0.5"
            ry="0.5"
            width="12"
            x="3"
            y="15"
          />
           
        </svg>
      </button>
    </span>
    <span
      class="ql-formats"
    >
      <button
        class="ql-list"
        title="Ordered List"
        type="button"
        value="ordered"
      >
        <svg
          viewBox="0 0 18 18"
        >
           
          <line
            class="ql-stroke"
            x1="7"
            x2="15"
            y1="4"
            y2="4"
          />
           
          <line
            class="ql-stroke"
            x1="7"
            x2="15"
            y1="9"
            y2="9"
          />
           
          <line
            class="ql-stroke"
            x1="7"
            x2="15"
            y1="14"
            y2="14"
          />
           
          <line
            class="ql-stroke ql-thin"
            x1="2.5"
            x2="4.5"
            y1="5.5"
            y2="5.5"
          />
           
          <path
            class="ql-fill"
            d="M3.5,6A0.5,0.5,0,0,1,3,5.5V3.085l-0.276.138A0.5,0.5,0,0,1,2.053,3c-0.124-.247-0.023-0.324.224-0.447l1-.5A0.5,0.5,0,0,1,4,2.5v3A0.5,0.5,0,0,1,3.5,6Z"
          />
           
          <path
            class="ql-stroke ql-thin"
            d="M4.5,10.5h-2c0-.234,1.85-1.076,1.85-2.234A0.959,0.959,0,0,0,2.5,8.156"
          />
           
          <path
            class="ql-stroke ql-thin"
            d="M2.5,14.846a0.959,0.959,0,0,0,1.85-.109A0.7,0.7,0,0,0,3.75,14a0.688,0.688,0,0,0,.6-0.736,0.959,0.959,0,0,0-1.85-.109"
          />
           
        </svg>
      </button>
      <button
        class="ql-list"
        title="Unordered List"
        type="button"
        value="bullet"
      >
        <svg
          viewBox="0 0 18 18"
        >
           
          <line
            class="ql-stroke"
            x1="6"
            x2="15"
            y1="4"
            y2="4"
          />
           
          <line
            class="ql-stroke"
            x1="6"
            x2="15"
            y1="9"
            y2="9"
          />
           
          <line
            class="ql-stroke"
            x1="6"
            x2="15"
            y1="14"
            y2="14"
          />
           
          <line
            class="ql-stroke"
            x1="3"
            x2="3"
            y1="4"
            y2="4"
          />
           
          <line
            class="ql-stroke"
            x1="3"
            x2="3"
            y1="9"
            y2="9"
          />
           
          <line
            class="ql-stroke"
            x1="3"
            x2="3"
            y1="14"
            y2="14"
          />
           
        </svg>
      </button>
    </span>
    <span
      class="ql-formats"
    >
      <button
        class="ql-script"
        title="Super"
        type="button"
        value="super"
      >
        <svg
          viewBox="0 0 18 18"
        >
           
          <path
            class="ql-fill"
            d="M15.5,7H13.861a4.015,4.015,0,0,0,1.914-2.975,1.8,1.8,0,0,0-1.6-1.751A1.922,1.922,0,0,0,12.021,3.7a0.5,0.5,0,1,0,.957.291,0.917,0.917,0,0,1,1.053-.725,0.81,0.81,0,0,1,.744.762c0,1.077-1.164,1.925-1.934,2.486A1.423,1.423,0,0,0,12,7.5a0.5,0.5,0,0,0,.5.5h3A0.5,0.5,0,0,0,15.5,7Z"
          />
           
          <path
            class="ql-fill"
            d="M9.651,5.241a1,1,0,0,0-1.41.108L6,7.964,3.759,5.349a1,1,0,1,0-1.519,1.3L4.683,9.5,2.241,12.35a1,1,0,1,0,1.519,1.3L6,11.036,8.241,13.65a1,1,0,0,0,1.519-1.3L7.317,9.5,9.759,6.651A1,1,0,0,0,9.651,5.241Z"
          />
           
        </svg>
      </button>
      <button
        class="ql-script"
        title="Sub"
        type="button"
        value="sub"
      >
        <svg
          viewBox="0 0 18 18"
        >
           
          <path
            class="ql-fill"
            d="M15.5,15H13.861a3.858,3.858,0,0,0,1.914-2.975,1.8,1.8,0,0,0-1.6-1.751A1.921,1.921,0,0,0,12.021,11.7a0.50013,0.50013,0,1,0,.957.291h0a0.914,0.914,0,0,1,1.053-.725,0.81,0.81,0,0,1,.744.762c0,1.076-1.16971,1.86982-1.93971,2.43082A1.45639,1.45639,0,0,0,12,15.5a0.5,0.5,0,0,0,.5.5h3A0.5,0.5,0,0,0,15.5,15Z"
          />
           
          <path
            class="ql-fill"
            d="M9.65,5.241a1,1,0,0,0-1.409.108L6,7.964,3.759,5.349A1,1,0,0,0,2.192,6.59178Q2.21541,6.6213,2.241,6.649L4.684,9.5,2.241,12.35A1,1,0,0,0,3.71,13.70722q0.02557-.02768.049-0.05722L6,11.036,8.241,13.65a1,1,0,1,0,1.567-1.24277Q9.78459,12.3777,9.759,12.35L7.316,9.5,9.759,6.651A1,1,0,0,0,9.65,5.241Z"
          />
           
        </svg>
      </button>
      <button
        class="ql-blockquote"
        title="Quote"
        type="button"
      >
        <svg
          viewBox="0 0 18 18"
        >
           
          <rect
            class="ql-fill ql-stroke"
            height="3"
            width="3"
            x="4"
            y="5"
          />
           
          <rect
            class="ql-fill ql-stroke"
            height="3"
            width="3"
            x="11"
            y="5"
          />
           
          <path
            class="ql-even ql-fill ql-stroke"
            d="M7,8c0,4.031-3,5-3,5"
          />
           
          <path
            class="ql-even ql-fill ql-stroke"
            d="M14,8c0,4.031-3,5-3,5"
          />
           
        </svg>
      </button>
      <button
        class="ql-link"
        type="button"
      >
        <svg
          viewBox="0 0 18 18"
        >
           
          <line
            class="ql-stroke"
            x1="7"
            x2="11"
            y1="7"
            y2="11"
          />
           
          <path
            class="ql-even ql-stroke"
            d="M8.9,4.577a3.476,3.476,0,0,1,.36,4.679A3.476,3.476,0,0,1,4.577,8.9C3.185,7.5,2.035,6.4,4.217,4.217S7.5,3.185,8.9,4.577Z"
          />
           
          <path
            class="ql-even ql-stroke"
            d="M13.423,9.1a3.476,3.476,0,0,0-4.679-.36,3.476,3.476,0,0,0,.36,4.679c1.392,1.392,2.5,2.542,4.679.36S14.815,10.5,13.423,9.1Z"
          />
           
        </svg>
      </button>
    </span>
  </div>
  <div
    class="quill "
  >
    <div
      class="ql-container ql-snow"
      style="position: relative;"
    >
      <div
        class="ql-editor"
        contenteditable="true"
        data-gramm="false"
        data-placeholder="Write something awesome..."
      >
        <p>
          test
        </p>
      </div>
      <div
        class="ql-clipboard"
        contenteditable="true"
        tabindex="-1"
      />
      <div
        class="ql-tooltip ql-hidden"
      >
        <a
          class="ql-preview"
          href="about:blank"
          rel="noopener noreferrer"
          target="_blank"
        />
        <input
          data-formula="e=mc^2"
          data-link="https://quilljs.com"
          data-video="Embed URL"
          type="text"
        />
        <a
          class="ql-action"
        />
        <a
          class="ql-remove"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`component renders without non required props 1`] = `
<div
  class="text-editor"
  data-testid="text-editor_test"
>
  <div
    class="ql-toolbar ql-snow"
    id="toolbar"
  >
    <span
      class="ql-formats"
    >
      <span
        class="ql-header ql-picker"
      >
        <span
          aria-controls="ql-picker-options-2"
          aria-expanded="false"
          class="ql-picker-label"
          data-label="Звичайний"
          data-value=""
          role="button"
          tabindex="0"
        >
          <svg
            viewBox="0 0 18 18"
          >
             
            <polygon
              class="ql-stroke"
              points="7 11 9 13 11 11 7 11"
            />
             
            <polygon
              class="ql-stroke"
              points="7 7 9 5 11 7 7 7"
            />
             
          </svg>
        </span>
        <span
          aria-hidden="true"
          class="ql-picker-options"
          id="ql-picker-options-2"
          tabindex="-1"
        >
          <span
            class="ql-picker-item"
            data-label="Заголовок"
            data-value="3"
            role="button"
            tabindex="0"
          />
          <span
            class="ql-picker-item"
            data-label="Вступ"
            data-value="4"
            role="button"
            tabindex="0"
          />
          <span
            class="ql-picker-item"
            data-label="Примітка"
            data-value="5"
            role="button"
            tabindex="0"
          />
          <span
            class="ql-picker-item ql-selected"
            data-label="Звичайний"
            data-value=""
            role="button"
            tabindex="0"
          />
        </span>
      </span>
      <select
        class="ql-header"
        style="display: none;"
      >
        <option
          value="3"
        >
          Заголовок
        </option>
        <option
          value="4"
        >
          Вступ
        </option>
        <option
          value="5"
        >
          Примітка
        </option>
        <option
          selected=""
          value=""
        >
          Звичайний
        </option>
      </select>
    </span>
    <span
      class="ql-formats"
    >
      <button
        class="ql-bold"
        title="Bold"
        type="button"
      >
        <svg
          viewBox="0 0 18 18"
        >
           
          <path
            class="ql-stroke"
            d="M5,4H9.5A2.5,2.5,0,0,1,12,6.5v0A2.5,2.5,0,0,1,9.5,9H5A0,0,0,0,1,5,9V4A0,0,0,0,1,5,4Z"
          />
           
          <path
            class="ql-stroke"
            d="M5,9h5.5A2.5,2.5,0,0,1,13,11.5v0A2.5,2.5,0,0,1,10.5,14H5a0,0,0,0,1,0,0V9A0,0,0,0,1,5,9Z"
          />
           
        </svg>
      </button>
      <button
        class="ql-italic"
        title="Italic"
        type="button"
      >
        <svg
          viewBox="0 0 18 18"
        >
           
          <line
            class="ql-stroke"
            x1="7"
            x2="13"
            y1="4"
            y2="4"
          />
           
          <line
            class="ql-stroke"
            x1="5"
            x2="11"
            y1="14"
            y2="14"
          />
           
          <line
            class="ql-stroke"
            x1="8"
            x2="10"
            y1="14"
            y2="4"
          />
           
        </svg>
      </button>
      <button
        class="ql-underline"
        title="Underline"
        type="button"
      >
        <svg
          viewBox="0 0 18 18"
        >
           
          <path
            class="ql-stroke"
            d="M5,3V9a4.012,4.012,0,0,0,4,4H9a4.012,4.012,0,0,0,4-4V3"
          />
           
          <rect
            class="ql-fill"
            height="1"
            rx="0.5"
            ry="0.5"
            width="12"
            x="3"
            y="15"
          />
           
        </svg>
      </button>
    </span>
    <span
      class="ql-formats"
    >
      <button
        class="ql-list"
        title="Ordered List"
        type="button"
        value="ordered"
      >
        <svg
          viewBox="0 0 18 18"
        >
           
          <line
            class="ql-stroke"
            x1="7"
            x2="15"
            y1="4"
            y2="4"
          />
           
          <line
            class="ql-stroke"
            x1="7"
            x2="15"
            y1="9"
            y2="9"
          />
           
          <line
            class="ql-stroke"
            x1="7"
            x2="15"
            y1="14"
            y2="14"
          />
           
          <line
            class="ql-stroke ql-thin"
            x1="2.5"
            x2="4.5"
            y1="5.5"
            y2="5.5"
          />
           
          <path
            class="ql-fill"
            d="M3.5,6A0.5,0.5,0,0,1,3,5.5V3.085l-0.276.138A0.5,0.5,0,0,1,2.053,3c-0.124-.247-0.023-0.324.224-0.447l1-.5A0.5,0.5,0,0,1,4,2.5v3A0.5,0.5,0,0,1,3.5,6Z"
          />
           
          <path
            class="ql-stroke ql-thin"
            d="M4.5,10.5h-2c0-.234,1.85-1.076,1.85-2.234A0.959,0.959,0,0,0,2.5,8.156"
          />
           
          <path
            class="ql-stroke ql-thin"
            d="M2.5,14.846a0.959,0.959,0,0,0,1.85-.109A0.7,0.7,0,0,0,3.75,14a0.688,0.688,0,0,0,.6-0.736,0.959,0.959,0,0,0-1.85-.109"
          />
           
        </svg>
      </button>
      <button
        class="ql-list"
        title="Unordered List"
        type="button"
        value="bullet"
      >
        <svg
          viewBox="0 0 18 18"
        >
           
          <line
            class="ql-stroke"
            x1="6"
            x2="15"
            y1="4"
            y2="4"
          />
           
          <line
            class="ql-stroke"
            x1="6"
            x2="15"
            y1="9"
            y2="9"
          />
           
          <line
            class="ql-stroke"
            x1="6"
            x2="15"
            y1="14"
            y2="14"
          />
           
          <line
            class="ql-stroke"
            x1="3"
            x2="3"
            y1="4"
            y2="4"
          />
           
          <line
            class="ql-stroke"
            x1="3"
            x2="3"
            y1="9"
            y2="9"
          />
           
          <line
            class="ql-stroke"
            x1="3"
            x2="3"
            y1="14"
            y2="14"
          />
           
        </svg>
      </button>
    </span>
    <span
      class="ql-formats"
    >
      <button
        class="ql-script"
        title="Super"
        type="button"
        value="super"
      >
        <svg
          viewBox="0 0 18 18"
        >
           
          <path
            class="ql-fill"
            d="M15.5,7H13.861a4.015,4.015,0,0,0,1.914-2.975,1.8,1.8,0,0,0-1.6-1.751A1.922,1.922,0,0,0,12.021,3.7a0.5,0.5,0,1,0,.957.291,0.917,0.917,0,0,1,1.053-.725,0.81,0.81,0,0,1,.744.762c0,1.077-1.164,1.925-1.934,2.486A1.423,1.423,0,0,0,12,7.5a0.5,0.5,0,0,0,.5.5h3A0.5,0.5,0,0,0,15.5,7Z"
          />
           
          <path
            class="ql-fill"
            d="M9.651,5.241a1,1,0,0,0-1.41.108L6,7.964,3.759,5.349a1,1,0,1,0-1.519,1.3L4.683,9.5,2.241,12.35a1,1,0,1,0,1.519,1.3L6,11.036,8.241,13.65a1,1,0,0,0,1.519-1.3L7.317,9.5,9.759,6.651A1,1,0,0,0,9.651,5.241Z"
          />
           
        </svg>
      </button>
      <button
        class="ql-script"
        title="Sub"
        type="button"
        value="sub"
      >
        <svg
          viewBox="0 0 18 18"
        >
           
          <path
            class="ql-fill"
            d="M15.5,15H13.861a3.858,3.858,0,0,0,1.914-2.975,1.8,1.8,0,0,0-1.6-1.751A1.921,1.921,0,0,0,12.021,11.7a0.50013,0.50013,0,1,0,.957.291h0a0.914,0.914,0,0,1,1.053-.725,0.81,0.81,0,0,1,.744.762c0,1.076-1.16971,1.86982-1.93971,2.43082A1.45639,1.45639,0,0,0,12,15.5a0.5,0.5,0,0,0,.5.5h3A0.5,0.5,0,0,0,15.5,15Z"
          />
           
          <path
            class="ql-fill"
            d="M9.65,5.241a1,1,0,0,0-1.409.108L6,7.964,3.759,5.349A1,1,0,0,0,2.192,6.59178Q2.21541,6.6213,2.241,6.649L4.684,9.5,2.241,12.35A1,1,0,0,0,3.71,13.70722q0.02557-.02768.049-0.05722L6,11.036,8.241,13.65a1,1,0,1,0,1.567-1.24277Q9.78459,12.3777,9.759,12.35L7.316,9.5,9.759,6.651A1,1,0,0,0,9.65,5.241Z"
          />
           
        </svg>
      </button>
      <button
        class="ql-blockquote"
        title="Quote"
        type="button"
      >
        <svg
          viewBox="0 0 18 18"
        >
           
          <rect
            class="ql-fill ql-stroke"
            height="3"
            width="3"
            x="4"
            y="5"
          />
           
          <rect
            class="ql-fill ql-stroke"
            height="3"
            width="3"
            x="11"
            y="5"
          />
           
          <path
            class="ql-even ql-fill ql-stroke"
            d="M7,8c0,4.031-3,5-3,5"
          />
           
          <path
            class="ql-even ql-fill ql-stroke"
            d="M14,8c0,4.031-3,5-3,5"
          />
           
        </svg>
      </button>
      <button
        class="ql-link"
        type="button"
      >
        <svg
          viewBox="0 0 18 18"
        >
           
          <line
            class="ql-stroke"
            x1="7"
            x2="11"
            y1="7"
            y2="11"
          />
           
          <path
            class="ql-even ql-stroke"
            d="M8.9,4.577a3.476,3.476,0,0,1,.36,4.679A3.476,3.476,0,0,1,4.577,8.9C3.185,7.5,2.035,6.4,4.217,4.217S7.5,3.185,8.9,4.577Z"
          />
           
          <path
            class="ql-even ql-stroke"
            d="M13.423,9.1a3.476,3.476,0,0,0-4.679-.36,3.476,3.476,0,0,0,.36,4.679c1.392,1.392,2.5,2.542,4.679.36S14.815,10.5,13.423,9.1Z"
          />
           
        </svg>
      </button>
    </span>
  </div>
  <div
    class="quill "
  >
    <div
      class="ql-container ql-snow"
      style="position: relative;"
    >
      <div
        class="ql-editor ql-blank"
        contenteditable="true"
        data-gramm="false"
        data-placeholder="Write something awesome..."
      >
        <p>
          <br />
        </p>
      </div>
      <div
        class="ql-clipboard"
        contenteditable="true"
        tabindex="-1"
      />
      <div
        class="ql-tooltip ql-hidden"
      >
        <a
          class="ql-preview"
          href="about:blank"
          rel="noopener noreferrer"
          target="_blank"
        />
        <input
          data-formula="e=mc^2"
          data-link="https://quilljs.com"
          data-video="Embed URL"
          type="text"
        />
        <a
          class="ql-action"
        />
        <a
          class="ql-remove"
        />
      </div>
    </div>
  </div>
</div>
`;
